### ISSUE TITLE
Changing strings with partial capo, pitch should stay the same

### ISSUE THREAD

**Issue by zacjansheski** [musescore/MuseScore#18450]
### Issue type

UI bug

### Bug description

When changing a string with partial capo, the pitch changes, it would be preferable for the pitch to stay the same.

### Steps to reproduce

1. Create a score with a tab instrument
2. Enable a partial capo
3. Enter a note and move it between strings (with and without capo)
4. Pitch changes when moving between strings

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/69917893/8bca23d9-e3e8-46e6-964f-b6184187ef13

### MuseScore Version

OS: macOS 11.6, Arch.: x86_64, MuseScore version (64-bit): 4.2.0-231861942, revision: github-musescore-musescore-279d82c

### Regression

No.

### Operating system

All

**Comment by scorster** [comment]
I'm interested in following the development of the MS 4.1 partial capo feature. Can someone point me to the beginning of that?

I've used the v4.1 partial capo function with great enthusiasm and posted some comments on the .org forum:

https://musescore.org/en/node/353203

Thanks! 

scorster

**Comment by bkunda** [comment]
Thanks @zacjansheski, I've raised some further thoughts for consideration here: https://www.dropbox.com/scl/fi/5d19x0x6jaxjadnjuuvrn/partial-capo-changing-strings.mov?rlkey=rfuclfhgjzte0d0onpuvltncj&dl=0

Would be interested to hear what @scorster has to say about this too. 

Thanks!

**Comment by zacjansheski** [comment]
@bkunda I agree, it makes more sense to me when thinking about the linked standard staff.  Feel free to close.

**Comment by sammik** [comment]
Sorry for long post.

I think, general concept of Capo and 8va is not right. 
It is wery well working workround, but not a correct implementation.
Why?

I think `pitch` property is "MIDI value", or "how note sounds", or in other words, tone frequency in Hz. (In real word, it depends on concert pitch and temperament, but for simplicity, we can ignore it for now and think defaults: A=440Hz, Equal Temperament)

This is correctly implemented in transposing instruments. `Pitch` remains unchanged, but `tpc` (where it is written) changes depending "concert pitch" vs "transposing (written) pitch".

But not in 8va and capos. 
In 8va there it is as "A4 (`pitch` 69) sounding as A5 (`pitch` 81). It looks correct:
![pitch](https://github.com/musescore/MuseScore/assets/1646034/96fd4409-3c98-4237-9229-6f9c1a0207dd)
But `pitch` is 69, and if `pitch` means "how note sounds", than "note sounding as 69 sounding as 81" is apparently non sense.

Correctly implementation, in my opinion, is "A5 (`pitch` 81) written in 3d gap".

Same in Capo. Example:
If I have melody in capoed Guitar 
![Capo-guit](https://github.com/musescore/MuseScore/assets/1646034/52762b5b-5aa9-4ee0-bd4b-4f4f35211f32)
And want to write same melody in piano, I would expect
![piano](https://github.com/musescore/MuseScore/assets/1646034/5b5da5e8-8afe-4636-a1eb-4b3a3466b996)

I am not saying, I dislike current behaviour. I like it, because it behaves as scordatura, which is one of my passions.
But I think, there should be always possibilty to see score in real (sounding) pitches: "what You see is what You hear".

I hope, it is understable, and hopefully makes a sense.

(May be engraving team has opinions on this)
